certificate/certificates.go

```diff
@@ -557,6 +610,69 @@ func (c *Certifier) Get(url string, bundle bool) (*Resource, error) {
 	}, nil
 }
 
+// GetRenewalInfo sends a request to the ACME server's renewalInfo endpoint to
+// obtain a suggested renewal window. The caller MUST provide the certificate
+// and issuer certificate for the certificate they wish to renew. The caller
+// should attempt to renew the certificate at the time indicated by the
+// ShouldRenewAt method of the returned RenewalInfoResponse object.
+//
+// Note: this endpoint is part of a draft specification, not all ACME servers
+// will implement it. This method will return api.ErrNoARI if the server does
+// not advertise a renewal info endpoint.
+//
+// https://datatracker.ietf.org/doc/draft-ietf-acme-ari
+func (c *Certifier) GetRenewalInfo(req RenewalInfoRequest) (*RenewalInfoResponse, error) {
+	certID, err := makeCertID(req.Cert, req.Issuer)
+	if err != nil {
+		return nil, fmt.Errorf("error making certID: %v", err)
+	}
+
+	resp, err := c.core.Certificates.GetRenewalInfo(certID)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
```

Comment:
Move this up to line 631 - right underneath where resp is declared, otherwise if the error case happens resp will be left open.

Reply:
The documentation for the HTTPClient indicates that the resp (*http.Response) will be `nil` when err is non-nil. This use of the HTTP client follows the same pattern followed elsewhere in this codebase.

Reply:
Ah, well today I learned.